### ISSUE TITLE
Unable to disable animated transitions on Material Top Tab Navigator

### ISSUE THREAD

**Issue by timsonater** [react-navigation/react-navigation#8273]
I am trying to disable animated transitions using screenOptions (just like on a stackNavigator as follows:

```
    const Tab = createMaterialTopTabNavigator();

    < Tab.Navigator
      screenOptions={{animationEnabled: false}}
    >
      < Tab.Screen name="Tab1" component={Tab1} />
      < Tab.Screen name="Tab2" component={Tab2} />
    </Tab.Navigator>
```

It is not disabling animation. I'm not positive this is a bug, is this supported? is there another way I should do this? Thanks! 

Using:
@react-navigation/core: "5.7.0",
@react-navigation/material-top-tabs: "5.2.5",

**Comment by carmenchapa** [comment]
Having the same problem here. Is there a way to achieve this?
`"@react-navigation/material-top-tabs": "^5.2.5"`

**Comment by raajnadar** [comment]
There is not option called as `animationEnabled`, read the docs carefully.

**Comment by timsonater** [comment]
So if my understanding is correct, this is impossible to achieve?

**Comment by andrewcharnley** [comment]
Same issue here.

**Comment by carmenchapa** [comment]
I opened a request in canny for this feature if you may upvote:
https://react-navigation.canny.io/feature-requests/p/allow-to-disable-animation-for-tabnavigator

If that's something already achievable it would be great know how.
I tried already with `cardStyleInterpolator` option, which works great for Stacks but it didn't with tabs

**Comment by ravirajn22** [comment]
Yes, its good to have feature. Generally the animated transition should only happen if you enable swipe gesture for switching tabs. Also if you have lots of tabs with contrasting content inside (say different colors, images etc), when you switch tab by pressing the tab, then UI feels little unpolished, you can see screens in between tabs while transition for microseconds. iOS uses different kind of animation transition for tab switch, which looks more modern. I will add those gifs when I have time.

**Comment by AyeshaIftikharr** [comment]
Hey Its not the very good solution and kind of a hack, but in order to disable the animations of the screens we can do the following for material top`Tab.Navigator` props:

```
<Tabs.Navigator
      initialRouteName='following'
      lazy
      timingConfig={{
        duration: 0, // will disable the animation
      }}
    >
      <Tabs.Screen
     ...
```

**Comment by sergeymorkovkin** [comment]
Applying timingConfig={{duration: 0}} dynamically is janky.

**Comment by amhouel** [comment]
I've tried timingConfig={{duration: 0}} on web and it makes the indicator buggy - I need to click on a tab twice to get the indicator to change. Does anyone know how to fix that or have a better solution to disable the animation?

**Comment by Gakki-coco** [comment]
> I've tried timingConfig={{duration: 0}} on web and it makes the indicator buggy - I need to click on a tab twice to get the indicator to change. Does anyone know how to fix that or have a better solution to disable the animation?

I tried 0.01, it is working now. XD

**Comment by vanhai472** [comment]
> Hey Its not the very good solution and kind of a hack, but in order to disable the animations of the screens we can do the following for material top`Tab.Navigator` props:
> 
> ```
> <Tabs.Navigator
>       lazy
>       timingConfig={{
>         duration: 0, // will disable the animation
>       }}
>     >
>       <Tabs.Screen
>      ...
> ```

this worked but the motion was 1 step too slow

**Comment by MichaelBorde** [comment]
The `timingConfig` hack won't work in `react-native-tab-view` v3 anymore.

See discussion here: https://github.com/satya164/react-native-tab-view/issues/1137

**Comment by bhaktitud** [comment]
I was also facing the same issue, where the client ask me to remove the transition animation in tab-view when user jump from tab-1 to tab-4 for instance. I was desperate because i could not found any working solution including the `timingConfig` hack. So, I decided to create the component my own just by using a simple `ScrollView` and `View` combination. Maybe if you still need it, you can use mind, just modify anything to fit your case. [https://github.com/bhaktitud/react-native-simple-tab-view](url)

**Comment by NicholasBoccuzzi** [comment]
Bumping this. Please allow us to disable the animation between top tabs.

**Comment by Scooter1337** [comment]
> This solution is based on the the comment [satya164/react-native-tab-view#1137 (comment)](https://github.com/satya164/react-native-tab-view/issues/1137#issuecomment-797421081). but patching the file mentioned doesn't work for me ,but patching `pagerViewAdapter.tsx` worked.
> 
> Tested on ios and android, it is working fine.
> 
> package name : react-native-tab-view
> 
> code in the patch file below:
> 
> ```
> diff --git a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
> index b858cb4..f715f5c 100644
> --- a/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
> +++ b/node_modules/react-native-tab-view/src/PagerViewAdapter.tsx
> @@ -63,7 +63,7 @@ export default function PagerViewAdapter<T extends Route>({
>        (route: { key: string }) => route.key === key
>      );
>  
> -    pagerRef.current?.setPage(index);
> +    pagerRef.current?.setPageWithoutAnimation(index);
>    }, []);
>  
>    React.useEffect(() => {
> @@ -72,7 +72,7 @@ export default function PagerViewAdapter<T extends Route>({
>      }
>  
>      if (indexRef.current !== index) {
> -      pagerRef.current?.setPage(index);
> +      pagerRef.current?.setPageWithoutAnimation(index);
>      }
>    }, [keyboardDismissMode, index]); 
> ```

Awesome! My tabbar is not updating anymore though.

**Comment by pierroo** [comment]
Sorry to be that guy but, is there any news on that?
Any clean way to handle disabling animation between tabs?

**Comment by ravirajn22** [comment]
This issue is fixed in this version using a new prop.
https://github.com/react-navigation/react-navigation/blob/main/packages/material-top-tabs/CHANGELOG.md#630-2022-10-11

Just need to be documented.

**Comment by DimitarTumbalov** [comment]
@avirajn22 is right! 
Adding `animationEnabled: false` to the **screenOptions**  of **Tab.Navigator** disables the animation.

**Comment by shovel-kun** [comment]
@timsonater Consider closing the issue since it has been resolved

**Comment by satya164** [comment]
Closing as this should be fixed in [d1bd31f](https://github.com/react-navigation/react-navigation/commit/d1bd31f11b1a40a319e7fb6e3fee074edd059349). Please upgrade to `@react-navigation/material-top-tabs@6.3.0` and give it a try.
